### ISSUE TITLE
Doesn't allow multiple commands for doctorTest

### ISSUE THREAD

**Issue by Maxim-Mazurok** [raineorshine/npm-check-updates#1259]
- [x] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)
- [x] I am using the latest version of `npm-check-updates`
- [x] I am using `node >= 14.14`

---

## Steps to Reproduce

I'm trying to run this:
```
npm-check-updates -u --doctor --doctorTest "npx -p typescript -c 'npm run test --ignore-scripts && tsc --noEmit'"
```

As you can see, I'm using npx with single quotes in order to run multiple commands in one go.

This is a simplified version that also doesn't work: `--doctorTest "npm test && tsc`

I understand that `npm test && tsc` doesn't work because here:
https://github.com/raineorshine/npm-check-updates/blob/HEAD/src/lib/doctor.ts#L117
We're splitting by `" "` and using first array element as a program to spawn.

And `--doctorTest "npx -p typescript -c 'npm run test --ignore-scripts && tsc --noEmit'"` doesn't seem to work because when we're splitting - we're not treating stuff in single quotes as a single argument.

A workaround, of course, is to just have an npm script that runs two commands, but in multi-root monorepo I'm trying to avoid duplication of that sort.

## Current Behavior

Doesn't spawn as expected

## Expected Behavior

It would be great if there was an option to specify two or more commands.

**Comment by Maxim-Mazurok** [comment]
Potential solution to extract arguments: https://regex101.com/r/2rzibO/1

**Comment by Maxim-Mazurok** [comment]
I've fixed the issue by applying the following patch with `npx patch-package`:

```diff
diff --git a/node_modules/npm-check-updates/build/src/lib/doctor.js b/node_modules/npm-check-updates/build/src/lib/doctor.js
index 15dfe45..debb95a 100644
--- a/node_modules/npm-check-updates/build/src/lib/doctor.js
+++ b/node_modules/npm-check-updates/build/src/lib/doctor.js
@@ -116,7 +116,15 @@ const doctor = async (run, options) => {
             },
         };
         if (options.doctorTest) {
-            const [testCommand, ...testArgs] = options.doctorTest.split(' ');
+            const regexp = /(?:'([^']+)'|([^ ]+))/g;
+            const matches = options.doctorTest.matchAll(regexp);
+            const groups = [];
+                
+            for (const match of matches) {
+              groups.push(match[1] === undefined ? match[2] : match[1]); // either first or second group
+            }
+            const [testCommand, ...testArgs] = groups;
+            console.log({testCommand, testArgs});
             await (0, spawn_please_1.default)(testCommand, testArgs, spawnOptions);
         }
         else {

```

I'd be happy to open a PR with this change if anyone finds this reasonable to add.

**Comment by Maxim-Mazurok** [comment]
My main concern was making it cross-platform without duplication, so creating a script probably won't easily satisfy my needs, same as running shell command (because shells are different on Win/Linux/Etc).
The way my solution works is by using `npx` (single process) that will run the following shell command: `npm test && tsc`.
Now I realize that my solution is also dependent on the system shell, so probably isn't 100% cross-platform, but luckily `&&` seems to work universally.
I'll create a PR for quotes support shortly.

**Comment by raineorshine** [comment]
Fixed and published in `v16.6.3`
